rfcs/NNNN-cdevents-provider/README.md

```diff
@@ -0,0 +1,173 @@
+within Tekton. 
+
+![User Stories Tekton](user-stories-provider.drawio.png)
+
+
+<!--
+Optional if existing discussions and/or issues are linked in the motivation section.
+-->
+
+### Alternatives
+
+There are currently no viable alternatives for the CDEvent Provider.
+<!--
+List plausible alternatives to the proposal and explain why the proposal is superior.
+
+This is a good place to incorporate suggestions made during discussion of the RFC.
+-->
+
+## Design Details
+
+Adding a Flux `Provider` for CDEvents that will send a CDEvent payload upon receiving a flux event from an alert. 
+
+Once an alert is triggered for this provider, it will send a CDEvent that corresponds to the Flux event that caused the
+alert. This CDEvent will be created using the [CDEvents Go SDK](https://github.com/cdevents/sdk-go). The implementation
+will include a defined mapping of which Flux events trigger a corresponding CDEvent.
+
+The CDEvents broker is not a part of this design and is left to the users to set up however they wish.
+
+Example Provider YAML:
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1beta2
+kind: Provider
+metadata:
+  name: cdevents-provider
+  namespace: flux-system
+spec:
+  type: cdevents
+  secretRef:
+    name: generic-token
+  address: http://10.100.0.17:5000
+```
+
+Example Default Mappings (Please note these are not final and will be open to discussion):
+
+| Helm Event                | CDEvent                         |
+| ------------------------- | ------------------------------- |
+| InstallSucceeded          | Environment.Modified            |
+|| Build.Finished (outcome not implemented yet) |
+|| PipeLineRun.Finished (with outcome success) |
+| InstallFailed             | Incident.Detected               |
+|| Build.Finished (outcome not implemented yet) |
+|| PipeLineRun.Finished (with outcome failed) |
+|UpgradeSucceeded           | TaskRun.Finished (with outcome success)                |
+|| Build.Finished (outcome not implemented yet) |
+|| PipeLineRun.Finished (with outcome success) |
+|UpgradeFailed              | TaskRun.Finished (with outcome failed)              |
+|| Build.Finished (outcome not implemented yet) |
+|| PipeLineRun.Finished (with outcome failed) |
+|TestSucceeded              | TestCaseRun.finished (outcome pass)           | 
+|TestFailed                 | TestCaseRun.finished (outcome fail)              | 
+|RollbackSucceeded          | TaskRun.Finished (with outcome success)           |
+|| Build.Finished (outcome not implemented yet) |
+|| PipeLineRun.Finished (with outcome success) |
+|RollbackFailed             | TaskRun.Finished (with outcome failed)           |
+|| Build.Finished (outcome not implemented yet) |
+|| PipeLineRun.Finished (with outcome failed) |
+|UninstallSucceeded         | Environment.Modified            |
+|UninstallFailed            | Incident.Detected               |
+|ReconciliationSucceeded    | Environment.Modified            | 
```

Comment:
This mapping feels wrong to me, we should map the HelmRelease to CDEvent `service` for the [deployed](https://github.com/cdevents/spec/blob/v0.4.1/continuous-deployment.md#service-deployed), [upgraded](https://github.com/cdevents/spec/blob/v0.4.1/continuous-deployment.md#service-upgraded), [rolledback](https://github.com/cdevents/spec/blob/v0.4.1/continuous-deployment.md#service-rolledback), [removed](https://github.com/cdevents/spec/blob/v0.4.1/continuous-deployment.md#service-removed) predicates.